pandas/_libs/hashtable_class_helper.pxi.in

```diff
@@ -416,7 +437,19 @@ cdef class {{name}}HashTable(HashTable):
                     labels[i] = count
                     count += 1
 
-        return np.asarray(labels)
+        return uniques.to_array(), np.asarray(labels)
+
+    def factorize(self, {{dtype}}_t[:] values):
+        return self._unique_with_inverse(values, uniques={{name}}Vector())
+
+    def get_labels(self, const {{dtype}}_t[:] values, {{name}}Vector uniques,
+                   Py_ssize_t count_prior=0, Py_ssize_t na_sentinel=-1,
+                   object na_value=None):
+        _, labels = self._unique_with_inverse(values, uniques,
+                                              count_prior=count_prior,
+                                              na_sentinel=na_sentinel,
+                                              na_value=na_value)
```

Comment:
Is it actually needed to still define this? From a quick search it seems `get_labels` was only used in `factorize` ?

Reply:
@jorisvandenbossche 
From a first try of uncommenting it, it wasn't straight-forward to remove. Definitely something for a follow-up. :)

Reply:
The only use case is in `_factorize_array`:

https://github.com/pandas-dev/pandas/blob/c8ce3d01e9ffafc24c6f9dd568cd9eb7e42c610c/pandas/core/algorithms.py#L475

Which can directly use the `hashtable.factorize` method. So I think this would be easy to remove, and would actually simplify further the diff.

Reply:
@jorisvandenbossche 
I'd prefer not loading up this PR with more things. The removal of `get_labels` in favour of `factorize` could be a separate PR, no?

Reply:
Yes, it could. But it is here that you are changing `get_labels`, and actually doing this would make the diff a lot smaller. So I think that is an indication that it naturally fits here.